src/material.cc

```diff
@@ -227,6 +227,24 @@ double Material::DecayHeat() {
   return decay_heat;
 }
 
+std::vector<Material::Ptr> Material::Package(Package::Ptr pkg) {
+  std::vector<Material::Ptr> ms_pkgd;
+  Material::Ptr m_pkgd;
+
+  double fill_mass = pkg->GetFillMass(quantity());
+  if (fill_mass == 0) {
+    return ms_pkgd;
+  }
+
+  while (quantity() > pkg->fill_min()) {
+    double pkg_fill = std::min(quantity(), fill_mass);
+    m_pkgd = boost::dynamic_pointer_cast<Material>(ExtractRes(pkg_fill));
```

Comment:
seems like this is the only line that really forces this method to be replicated 3 times... I'll have to dust off my C++ polymorphism to see if there's a better way...